### PR TITLE
[108, 87] Close Mentorship (endpoint, button handle, email notifications)

### LINKED ISSUE #87: Close mentorship: email notifications
### Dependency

This depends on #108 closed.

### Problem

In #108 we allow mentor/mentee to close a mentorship releationship.

Currently there is no email sent as notification.

### Feature Description

- add sending email notification calling the existing function `sendMentorMentorshipClosedEmail` in `utils/email/index.js` from the  `closeMentorship` API endpoint, similarly to https://github.com/tupuio/tupuapp/pull/93

### LINKED ISSUE #108: Close Mentorship (endpoint, button handle)
## Dependencies

- #107 
 
### Feature Description

When a mentor/mentee clicks on the "Close Mentorship" button of #107, we want to have:
1. a confirmation message box dialog with user feedback (toast, button loading): same as in #78
2. a `closeMentorship` API endpoint to close the mentorship, and remove it from the active ones (status -> closed) similarly to #65
3. remember to follow up with #87

### PR DESCRIPTION
**Pull request by lorepirri**
### What Github issue does this PR relate to?

Close #108 
Close #87 

Changes proposed in this pull request:

- endpoint, button handle, email notifications

### What should the reviewer know?

the handle for the button to close the mentorship is temporary, and it should be moved into #107 

### Quality Assurance(QA) Checklist

- [x] The app builds and runs properly

### CODE REVIEW

**Comment by lorepirri** [comment]
ready to be reviewed

File: components/MentorshipsList/MentorshipCard.js [hunk 1]
```diff
@@ -87,6 +87,28 @@ const MentorshipCard = ({ mentorship }) => {
...
+            <MenuList>
+              <MenuItem 
+                onClick={() => handleCloseMentorship(mentorship, RelationshipStatusEnum.ClosedCompleted)}
+              >Completed</MenuItem>
+              <MenuItem 
+                onClick={() => handleCloseMentorship(mentorship, RelationshipStatusEnum.ClosedNoGoodFit)}
+              >Not a good fit</MenuItem>
+              <MenuItem 
+                onClick={() => handleCloseMentorship(mentorship, RelationshipStatusEnum.ClosedBusy)}
+              >I&apos;m busy</MenuItem>
```
Comment (raluka) [inline components/MentorshipsList/MentorshipCard.js:106]:
```suggestion
              >Not available</MenuItem>
```

File: components/MentorshipsList/index.js [hunk 2]
```diff
@@ -1,12 +1,92 @@
...
 const MentorshipsList = () => {
-  const { data, error } = useSWR("/api/mentorships", fetcher);
+  const { data, error, mutate } = useSWR("/api/mentorships", fetcher);
+
+  const [actionDialogParams, setActionDialogParams] = useState(null);
+  const [isActionDialogLoading, setActionDialogLoading] = useState(false);
+  const { isOpen: isAlertDialogOpen, onOpen: onAlertDialogOpen, onClose: onAlertDialogClose } = useDisclosure();
+  const cancelRef = useRef();
+  const toast = useToast();
+
```
Comment (maranta1) [inline components/MentorshipsList/index.js:20]: Should we move the `mutateRelationshipsList` and `changeMentorshipRelationship` functions into a separate file? At the moment that will be declared again each time the component renders

**Review by maranta1 — APPROVED** [review]
🚀

File: components/MentorshipsList/index.js (hunk 2 above)
Comment (lorepirri) [reply components/MentorshipsList/index.js:20]: I created a follow up ticket: https://github.com/tupuio/tupuapp/issues/118

**Comment by lorepirri** [comment]
thanks @raluka and @maranta1 for reviewing this! I am merging it 👍

**Pull request merged**